### PR TITLE
Fix #6923/#6926: Sidebar modal and showCloseIcon attrs

### LINKED ISSUE #6923: Sidebar: add attribute to disable modal masking
The Sidebar always darkens the background when opened. There should be a `modal` attribute to set the masking (like there is for the Dialog component).

### LINKED ISSUE #6926: Sidebar: add attribute to disable close on click
The Sidebar always closes when clicking anywhere on the background.
There should be an option to disable this behaviour.
There could also be a `permanent` attribute to show the sidebar permanently.

### PR DESCRIPTION
**Pull request by melloware**
Following OverlayPanel lead added 2 new attributes `modal` and `showCloseIcon` to follow it.

### CODE REVIEW

File: src/main/java/org/primefaces/component/sidebar/SidebarRenderer.java [hunk 1]
```diff
@@ -91,9 +93,11 @@ private void encodeScript(FacesContext context, Sidebar bar) throws IOException
         WidgetBuilder wb = getWidgetBuilder(context);
         wb.init("Sidebar", bar)
                 .attr("visible", bar.isVisible(), false)
+                .attr("modal", bar.isModal(), false)
```
Comment (tandraschko) [inline src/main/java/org/primefaces/component/sidebar/SidebarRenderer.java]: false? shouldnt it be true?
This flag is actually used to NOT pass the default value to the JS.
the JS then needs to interpret "null" as "true" of course

Comment (melloware) [reply src/main/java/org/primefaces/component/sidebar/SidebarRenderer.java]: Yep I was being lazy let me fix.

**Comment by melloware** [comment]
Updated for proper defaults.

**Comment by tandraschko** [comment]
looks good now
+1 if well tested

**Pull request merged**